src/snowflake/connector/result_set.py

```diff
@@ -178,19 +179,26 @@ def _fetch_pandas_batches(self, **kwargs) -> Iterator[DataFrame]:
         Snowflake's back-end.
         """
         self._can_create_arrow_iter()
-        return self._create_iter(iter_unit=IterUnit.TABLE_UNIT, structure="pandas")
+        return self._create_iter(
+            iter_unit=IterUnit.TABLE_UNIT, structure="pandas", **kwargs
+        )
 
     def _fetch_pandas_all(self, **kwargs) -> DataFrame:
         """Fetches a single Pandas dataframe."""
-        dataframes = list(self._fetch_pandas_batches())
+        concat_args = list(inspect.signature(pandas.concat).parameters)
```

Comment:
I think we could get away with something simpler, but this will work. Hopefully `pandas.concat` will never introduce an argument that conflicts with our `to_pandas` function